### ISSUE TITLE
canGetAndSetFMUstate support for exported FMU

### ISSUE THREAD

**Issue by Midren** [OpenModelica/OpenModelica#7939]
### Description

Currently is it is not possible to FMU with enabled canGetAndSetFMUstate flag, by exporting from OpenModelica.

### Expected Behavior

Some possibility to export FMU with enabled canGetAndSetFMUstate.

**Comment by bilderbuchi** [comment]
Cross-ref where lack of this is relevant when using FMPy: https://github.com/CATIA-Systems/FMPy/issues/303#issuecomment-891043884

**Comment by lochel** [comment]
We started with a first prototype implementation in #8638.

**Comment by sturmk** [comment]
> We started with a first prototype implementation in #8638.

Are there examples or tests for this feature? fmi2GetFMUstate keeps SEGFAULT'ing for me right before the function returns.

**Comment by arun3688** [comment]
@sturmk here is an example below with .mos script

a.mos

```
setCommandLineOptions("-d=newInst,-disableDirectionalDerivatives,fmuExperimental");
getErrorString();

loadString("
model fmi_attributes_18
  input Real u1(start=1);
  input Real u2(start=2);
  output Real y;
  Real x;
initial equation
  x = u1*u2 + (u1 + u2)/2;
equation
  der(x) = x*u1^2;
  y = x + u1*u2 + x;
end fmi_attributes_18;
"); getErrorString();

translateModelFMU(fmi_attributes_18, version="2.0", fmuType="me_cs"); getErrorString();
```

You have to run the following command

`>>> omc a.mos`

**Comment by awecefil** [comment]
Hi, is this feature only supports Co-Simulation? I add `-d=fmuExperimental` in OMEdit > Simulation Setup > Translation Flags > Additional Translation Flags, but the exported FMU show that only Co-Simulation's Get/Set State is true
![image](https://github.com/OpenModelica/OpenModelica/assets/16224322/d04c1bc5-67b3-46e8-a5fb-ae740bafaaed)

**Comment by casella** [comment]
@arun3688 can you please comment on that?

**Comment by bilderbuchi** [comment]
> Hi, is this feature only supports Co-Simulation? I add `-d=fmuExperimental` in OMEdit > Simulation Setup > Translation Flags > Additional Translation Flags, but the exported FMU show that only Co-Simulation's Get/Set State is true ![image](https://github.com/OpenModelica/OpenModelica/assets/16224322/ecb9293e-1539-4634-8dc0-fb4bfc9c6f00)

I just exported an ME FMU with the flag activated, on OpenModelica v1.25.0-dev-272-gfe7f1f9d6c , and I successfully get the expected entry in the modeldescription.xml:
>    canGetAndSetFMUstate="true"
    canSerializeFMUstate="true"

So it seems that might have landed in the meantime?

**Comment by bilderbuchi** [comment]
Is it documented when the `canGetAndSetFMUstate` capability transitions out of the "fmuExperimental" flag into support by default, or what is blocking that transition?

**Comment by casella** [comment]
@arun3688 can you please comment on why the fmi2GetSpecificDerivatives, canGetSetFMUState, canSerializeFMUstate options are still considered as experimental?

**Comment by arun3688** [comment]
@casella @bilderbuchi @lochel When we did this work , it was a prototype and we did not do extensive testing, And it might affect the library coverage i guess so that is why it is still experimental. But now it has been in the code for more than 3 years and people are using it and we did not get any issues on this feature @lochel can you please comment on this, should we make this as default ?

**Comment by sturmk** [comment]
I hope you don't mind me chiming in; I still see two major issues that will cause a headache if more people use it: #10975 #10790

**Comment by lochel** [comment]
Yes, this feature isn't tested extensively, and it probably never will be if marked as experimental. It would be good if we could check it's impact on the coverage test.
